tests/binning_test.py

```diff
@@ -547,3 +548,189 @@ def test_variable_bin_equivalent_to_bin_of_data_array_with_counts():
+    da = table.bin(x=size * size)
+    x1 = sc.arange('x', size * size) // size
+    x2 = sc.arange('x', size * size) % size
+    da.coords['x1'] = x1
+    da.coords['x2'] = x2
+    # The old "standard" implementation uses a massive amount of memory, crashing even
+    # with 256 GByte of RAM.
+    assert da.group('x1', 'x2').dims == ('x1', 'x2')
+
+
+def test_group_many_and_erase():
+    from scipp import binning
+    binning.make_binned
+    size = 512
+    table = sc.data.table_xyz(int(1e3))
+    da = table.bin(z=3, x=size, y=size)
+    xcoarse = sc.arange('x', size) // 2
+    ycoarse = sc.arange('y', size) // 2
+    da.coords['xcoarse'] = xcoarse
+    da.coords['ycoarse'] = ycoarse
+    result = binning.make_binned(
+        da,
+        edges=[],
+        groups=[sc.arange('xcoarse', size // 2, dtype=xcoarse.dtype)],
+        erase=['x', 'y'])
+    assert result.dims == ('z', 'xcoarse')
+
+
+def test_erase_multiple():
+    from scipp import binning
+    binning.make_binned
+    table = sc.data.table_xyz(int(1e3))
+    da = table.bin(x=13, y=7)
+    result = binning.make_binned(da, edges=[], groups=[], erase=['x', 'y'])
+    assert sc.identical(result.value, da.bins.constituents['data'])
+
+
+def test_erase_2d_mask():
+    from scipp import binning
+    binning.make_binned
+    table = sc.data.table_xyz(100)
+    table.data = sc.arange('row', 100)
+    da = table.bin(x=2, y=2)
+    da.masks['mask'] = sc.zeros(dims=da.dims, shape=da.shape, dtype=bool)
+    da.masks['mask'].values[1, 1] = True
+    result = binning.make_binned(da, edges=[], groups=[], erase=['x', 'y'])
+    assert not sc.identical(result.sum(), table.sum())
+    assert sc.identical(result.sum(), da.sum())
+
+
+def test_erase_multiple_masks():
+    from scipp import binning
+    binning.make_binned
+    table = sc.data.table_xyz(100)
+    table.data = sc.arange('row', 100)
+    da = table.bin(x=2, y=2)
+    da.masks['xy'] = sc.zeros(dims=da.dims, shape=da.shape, dtype=bool)
+    da.masks['xy'].values[1, 1] = True
+    da.masks['x'] = sc.array(dims=['x'], values=[False, True])
+    result = binning.make_binned(da, edges=[], groups=[], erase=['x', 'y'])
+    assert not sc.identical(result.sum(), table.sum())
+    assert sc.identical(result.sum(), da.sum())
+
+
+def _make_base_array():
+    nx = 7
+    ny = 3
+    nz = 13
+    sizes = {'x': nx, 'y': ny, 'z': nz}
+    nrow = 10000
+    table = sc.data.table_xyz(nrow)
+    table.data = sc.array(dims=['row'], values=rng.integers(1000000, size=nrow))
+    da = table.bin(sizes)
+    da.coords['x1'] = sc.arange('x', nx) // 2
+    da.coords['x2'] = sc.array(dims=['x'], values=rng.integers(nx // 3, size=nx))
+    da.coords['y1'] = sc.arange('y', ny) // 2
+    da.coords['y2'] = sc.array(dims=['y'], values=rng.integers(ny // 3, size=ny))
+    da.coords['z1'] = sc.arange('z', nz) // 2
+    da.coords['z2'] = sc.array(dims=['z'], values=rng.integers(nz // 3, size=nz))
+    return da
+
+
+_base_array = _make_base_array()
```

Comment:
I wanted to avoid making a new array every time since that slows down the tests (due to the many combinations/permutations).

Reply:
Ok. I don't think this can be avoided nicely with fixtures. It would still be possible to user `mark.parametrize` with a generator which would call `_make_base_array` and `cases` internally. But I am not sure how complicated that would get.